### DISCUSSION TITLE
Can't get rid of "nvim-treesitter[yaml]: Error during compilation" on startup

### DISCUSSION THREAD

**Discussion by mkastner** [LazyVim/LazyVim#1967] [General]
I have a fresh LazyVim install and I keep getting the error message below on startup. It's got most likely nothing to with LazyVim. It does not happen with my self made init.lua which also uses treesitter. 

I probably need to ask for help at the nvim-treesitter repository. But since this error only occured at a new LazyVim install, I thougth I ask here first.

I know, it has to do with this -macosx_version_min having changed to -macos_version_min. Maybe someone had a similar problem and was able to fix it:

```
nvim-treesitter[yaml]: Error during compilation
-macosx_version_min has been renamed to -macos_version_min
0  0x10ba53f43  __assert_rtn + 64
1  0x10b955f43  ld::AtomPlacement::findAtom(unsigned char, unsigned long long, ld::AtomPlacement::AtomLoc const*&, long long&) const + 1411
2  0x10b972431  ld::InputFiles::SliceParser::parseObjectFile(mach_o::Header const*) const + 19745
3  0x10b982b71  ld::InputFiles::parseAllFiles(void (ld::AtomFile const*) block_pointer)::$_7::operator()(unsigned long, ld::FileInfo const&) const + 657
4  0x7ff80f78f5cd  _dispatch_client_callout2 + 8
5  0x7ff80f79fe3e  _dispatch_apply_invoke + 214
6  0x7ff80f78f59a  _dispatch_client_callout + 8
7  0x7ff80f79e99d  _dispatch_root_queue_drain + 879
8  0x7ff80f79ef22  _dispatch_worker_thread2 + 152
9  0x7ff80f933c06  _pthread_wqthread + 262
ld: Assertion failed: (resultIndex < sectData.atoms.size()), function findAtom, file Relocations.cpp, line 1336.
collect2: error: ld returned 1 exit status
```

**Comment by dpetka2001** [comment]
Not using MacOS myself, but found this [post](https://github.com/ikatyang/tree-sitter-yaml/issues/53). Maybe it'll prove relevant and help you out with your problem? But I agree that if you don't manage to solve it yourself, a more appropriate approach would be an issue at treesitter's repo for better guidance to your issue. @amaanq Sorry for pinging you, but you're the only one that I know of who's involved with nvim-treesitter. Maybe you have a more helpful suggestion regarding this issue?

**Reply by mkastner** [reply]
@dpetka2001 I appreciate your helpful consideration and pinging someone from the treesitter project. Thanks, very kind!

**Reply by dpetka2001** [reply]
You should also try the solution mentioned in the post i linked to. Maybe it will solve your problem?

**Reply by mkastner** [reply]
Now, I have – kaboum ... it worked. Sorry, for letting you wait for a feedback. Got a call and then I tried to make sense. Well anyway, your link is pure gold. 
I'm still scratching my head on using the zig compiler. I'm currently doing a Zig project myself. I knew, Zig compiles C like a charm, but I'm on nightly i.e. master and that worked too.
I'm deeply grateful, seriously!

**Reply by dpetka2001** [reply]
Glad you got it working. Not familiar with MacOS or Zig myself, so all I could do was provide some hopefully helpful link by searching for similar problems.

**Reply by amaanq** [reply]
> @amaanq Sorry for pinging you, but you're the only one that I know of who's involved with nvim-treesitter. Maybe you have a more helpful suggestion regarding this issue?

No worries at all, as long as it's not an issue that's been posted 100x :grin: 

My guess is Apple's clang is linking against the wrong path for the stdlib/include paths, and imo Apple clang sorta sucks, can you install LLVM's clang with `brew install clang` and use that as your compiler? 

If Zig works for you then more power to you, but if you prefer clang this might work :slightly_smiling_face:
